docs/02-guides/server-side-events/create-an-event.mdx

```diff
@@ -1,15 +1,135 @@
 ---
 title: Create an event
 description:
-  This guide explains how to create an event in Front-Commerce and dispatch it to external services.
+  In this guide, we will cover the creation of an emitter and how to implement
+  them in your pages, we will create an Emitter that Emit an event when a
+  customer votes on an FAQ question.
 sidebar_position: 1
 ---
 
 <p>{frontMatter.description}</p>
 
-:::info WIP
+## What is an event emitter
 
-- Create an event emitter
-- Emit an event from your code
+The events are dispatched through the use of an event emitter. The role of the
+event emitter is to push events to the server event pipeline to be processed by
+the server.
+
+## Prerequisites
+
+Before diving into the code, ensure you have the following prerequisites in
+place:
+
+- A Front-Commerce project configured and ready to use
+
+:::info
+
+The outcome of this guide is already available in the
+[FAQ example extension](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/tree/main/skeleton/example-extensions/faq-demo).
+
+:::
+
+## Create a Domain event class
+
+Let's create a file in our project, called `QuestionVotedEvent.ts`:
+
+```tsx title="app/server-events/events/QuestionVotedEvent.ts"
+import { ServerEvent } from "@front-commerce/core/events";
+
+const QuestionVotedEventName = "QuestionVotedEvent";
+
+// This is the payload data that we want to gather for the event
+type QuestionVotedEventPayload = {
+  questionSlug: string;
+  upvote: boolean;
+};
+
+// We define an event class that extends the ServerEvent class
+// This class will handle data to be pushed to the server event pipeline
```

Comment:
**thought:** "handle" can be misleading here, in the sense it may mean the class may alter the data. However, this is a domain class, meaning it's basically a data container.

**suggestion:** Change "handle" to "contain".